### ISSUE TITLE
Unit Query Panel bugs

### ISSUE THREAD

**Issue by fgarbe** [jeffhill/Dota2#333]
**I am aware that the unit query panel is no longer officially maintained, but I still want to try and revive that feature.**

Objectively it offers you the ability to keep control / sight on your units as well as view information about a second unit and is a standard in other games such as Heroes of Newerth, Heroes of the Storm or League of Legends.

Bugs I've noticed:

1. Cannot see allied or enemy tp scroll slot
2. Cannot see allied or enemy neutral item slot
3. Cannot see teammates quickbuy, this could be added in some fashion (with proper borders and separators) like this: 
![image](https://user-images.githubusercontent.com/1477819/171796707-02116336-969f-40e2-8506-a33be7a14695.png)

4. Cannot see Roshan's inventory. This is pretty important to know whether he'll drop Cheese or Refresher shard or Aghanim's etc.
5. Cannot see targets benefits from main attribute, e.g. Status Resistance for Strength heroes.
6. No channeling cast bar.
7. Unit query window overlaps with icons for the player's & teammates' voice chat indicators.
8. Portrait is always animated even if the setting for player portrait animation is turned off.
9. Double clicking the portrait initiates a camera lock, that is not broken when trying to center your hero. You have to use ESC to get rid of the camera lock or wait 8 seconds.
10. Remove the sliding transition animation when initially clicking on a unit. The sliding animation has been removed from the scoreboard but is still present for the Unit Query window.
11. Venomancer's Plague Wards do not show any HP:
![image](https://user-images.githubusercontent.com/1477819/171796414-e2932dbe-3320-4713-bec2-e31d83a996b7.png)
12. Selecting a spirit bear looks very weird:
![image](https://user-images.githubusercontent.com/1477819/171796447-623a2373-5a41-4e6b-9d03-b405e4df8f69.png)
13. Selecting an infested Naix looks very weird as well
14. Tombstone is bugged aswell when selected
15. Cannot see selected heroes Aghanim Shard/Blessing status

**Comment by NMOTH** [comment]
#135, #471

**Comment by Aluerie** [comment]
This thread has the most upvotes from existing issues about UnitQuery panel so it's probably better to send people thumb up /comment this one in future...

So I gonna add: 

15. Cannot see enemy heroes eaten Aghanim Shard/Blessing

**Comment by faddeefde** [comment]
I gonna add too:
  
  16. Remove the hero's name, it's just useless

**Comment by unitqueryonepercenter** [comment]
17, a better name for the option rather than "unit query overrides hero control".

Honestly the name doesn't matter but I'd love to see this get the update it deserves!

**Comment by bwanners** [comment]
I only started using the side panel recently but think it is such a great idea and sad to see that it isn't maintained anymore. All the features you mentioned are so critical to gameplay. Hope it gets more traction. Thanks!

**Comment by jeffhill** [comment]
BLUF: We're not maintaining this panel and we are planning to remove it after TI this year.

Thank you for taking the time to report these bugs and thank you to everyone who upvoted the issue! This issue led to a broader discussion this morning in the Dota developers chat about what to do about this issue - but we still don't plan to support this feature long term. This panel was a neat idea we tried in the UI redesign some years ago and while it was kind of interesting, in practice very few players actually used the panel even when it was new. Since we're going to remove this panel in a few months time, and so few players are using it today, it doesn't make a lot of sense to spend the time fixing these bugs with it.

I'm sorry this is the response we've got for you on this, but your issue deserves an honest and complete answer even if it's probably not the one you're hoping for.

**Comment by anko** [comment]
@jeffhill 

> in practice very few players actually used the panel even when it was new [...] and so few players are using it today

*The panel has never even shown the TP slot, or Roshan's inventory!*

If you release a feature broken and never update it, and then observe that nobody wants to use the obviously broken and unsupported feature, you can't logically conclude that nobody must have wanted it.

That's like testing whether people prefer bread with rotten butter on it, observing that nobody likes rotten-butter-bread, and concluding that bread is best served plain and butter was a mistake.  Perhaps you should try fresh butter instead?

**Comment by AlynxZhou** [comment]
This is a basic feature for me, because if I click other unit by mistake, I can still use my own ability with this panel. But if I don't have this detached panel, my own unit is covered, I must first re-select myself, and only then can I use my own ability or item.

Without this panel, I always select other unit by mistake (because my slow reaction, I always repeat "press ability, then press another hero" many times to make sure I can use it at the first time, and if those quick actions are not in sequence, I get other hero selected), and then I forget to re-select myself and do nothing and die. This side panel does not block my own ability, so even if I select other hero wrongly, I can still use my own abilities or items.

If you really decide to remove it, please consider to implement something like "if I select another unit but I cannot use it's abilities or tools, I'll get myself re-selected automatically when I press keys for abilities or tools".

Maybe many people have fast reaction so they can manually select themselves quickly if they click other units by mistake, but for a slow reaction guy like me, it's really hard. Maybe you'll say "is it so hard to press one more F1?", but it is just so hard for me, just like my fast reaction friends always ask "why you can't use your magic wand/eul/atos at that time?", maybe that's easy for many people, but it's hard for slow reaction guys. I even bind "select myself" to MOUSE4 to make it easier to press, but I still fail on this (I mean, re-selecting myself after selecting other units by mistake). I don't want to be slow reaction, either, but that is not something decided by myself, I tried a lot and found I really need this side panel to enjoy a normal game and help my teammates with my abilities.

Yes, I can even accept some trade-offs like no TP, no neutral item slot, just because it's hard for me to use abilities without this panel.

**Comment by bwanners** [comment]
> BLUF: We're not maintaining this panel and we are planning to remove it after TI this year.
> 
> Thank you for taking the time to report these bugs and thank you to everyone who upvoted the issue! This issue led to a broader discussion this morning in the Dota developers chat about what to do about this issue - but we still don't plan to support this feature long term. This panel was a neat idea we tried in the UI redesign some years ago and while it was kind of interesting, in practice very few players actually used the panel even when it was new. Since we're going to remove this panel in a few months time, and so few players are using it today, it doesn't make a lot of sense to spend the time fixing these bugs with it.
> 
> I'm sorry this is the response we've got for you on this, but your issue deserves an honest and complete answer even if it's probably not the one you're hoping for.

@jeffhill 
Exactly what everyone else is saying. This is the preferred way of playing for a lot of people. And it may be, as percentage-wise, nowhere near the benchmark that you'd like to see, but people do in fact use it and love it. As others mention, pro players use it. Being a fairly new player myself (about 2 years) I never knew about it until watching OG.Taiga stream and I asked him about that "plug-in" and he simply said it was a built in option. None of my friends knew about this either. It isn't marketed at all and is in the most random location in settings that if you are not curious, you would never find it. I'm sure a lot of people would use it if 1) it was more well known and 2) it was updated and fixed.

I get it, resources are always limited. But please at the very minimum do not get rid of the feature. A lot of people are just used to playing this way now and prefer it. You may not update it, but at the very least please do not get rid of it. My two cents. Thank you.

**Comment by unitqueryonepercenter** [comment]
@jeffhill 
What a slap to the face. This is a numbers-driven bug tracker, you've expressed again and again that you want to work on what the community thinks is important, namely QoL changes that don't break the game.

This is one of the highest voted topics on this forum?? think of the amount of player that aren't even aware of this tracker, that still long for the update this so sorely deserves. It just doesn't compute Jeff.

Maybe the software you use to determine the amount of unit query users is broken too?

**Comment by unitqueryonepercenter** [comment]
and outright removing it? seriously, what a tone deaf response.
like wow, sorry for bringing this issue to your attention. maybe if we just shut up we'd still get the broken version.
I hope all your games are ruined by smurfs jeff.

**Comment by anko** [comment]
For posterity: @jeffhill explained the decision further in [a Reddit comment](https://www.reddit.com/r/DotA2/comments/vig0uw/jeffhill_regarding_the_unit_query_panels/ide4htv/).

**Comment by unitqueryonepercenter** [comment]
https://www.reddit.com/r/DotA2/comments/vig0uw/jeffhill_regarding_the_unit_query_panels/ide4htv/

tldr: it's hard to do...

doesn't matter that it's been one of the most requested fixes since it fell behind - that shit takes time to fix lmao

**Comment by unitqueryonepercenter** [comment]
> Would you rather they fix this UI, than get Battlepass out on time?
yes?
but alas, no time, no money. just a humble indie company with only one dev. I guess I have to manage my expectations because gee wizz you've all made it clear that its super dooper hard to do :((

**Comment by MonsieurPi** [comment]
If I can just drop my 2 cents here, @jeffhill , please don't delete it at least, even if incomplete I'm happy with my current setting where I bound a key to toggle the Unit Query Panel on and off with:

```
bindtoggle "F1" dota_hud_new_query_panel
```

This way I can at least not have my panel replaced by other units panels most of the time and, when needed, I simply toggle it to check their neutral items, tp etc

**Comment by xDekaLx** [comment]
#17511

**Comment by mattiasdrp** [comment]
Did you remove it in 7.39? It's not there anymore. 

I still don't get why you maintain legacy keys (even for new heroes that have no legacy bound to them) but never tried to revamp the side query panel (even by adding it to the dota labs so people can try it)
